backend/manager/modules/utils/src/main/java/org/ovirt/engine/core/utils/ovf/OvfReader.java

```diff
@@ -975,6 +978,44 @@ private void fixDiskVmElements() {
         }
     }
 
+    protected void readNumaNodeListSection(XmlNode section) {
+        XmlNodeList list = selectNodes(section, NUMA_NODE);
+        List<VmNumaNode> vmNumaNodes = new ArrayList<>();
+        vmBase.setvNumaNodeList(vmNumaNodes);
+
+        for (XmlNode node : list) {
+            VmNumaNode vmNumaNode = new VmNumaNode();
+            XmlNode id = selectSingleNode(node, "id", _xmlNS);
+            if (id != null) {
+                vmNumaNode.setId(new Guid(id.innerText));
+            }
+            vmNumaNode.setIndex(Integer.valueOf(selectSingleNode(node, NUMA_INDEX, _xmlNS).innerText));
+            vmNumaNode.setCpuIds(readIntegerList(node, NUMA_CPU_ID_LIST));
+            vmNumaNode.setVdsNumaNodeList(readIntegerList(node, NUMA_VDS_NUMA_LIST));
+            vmNumaNode.setMemTotal(Long.valueOf(selectSingleNode(node, NUMA_TOTAL_MEMORY, _xmlNS).innerText));
+            XmlNode numaTuneMode = selectSingleNode(node, NUMA_TUNE_MODE, _xmlNS);
+            if (numaTuneMode != null) {
+                vmNumaNode.setNumaTuneMode(NumaTuneMode.forValue(numaTuneMode.innerText));
+            }
+            vmNumaNodes.add(vmNumaNode);
+        }
+    }
+
+    private List<Integer> readIntegerList(XmlNode node, String label) {
```

Comment:
how about extracting this to OvfUtils? seems it only needs to get _xmlNS

Reply:
I don't think it worth it. It also needs moving the `selectSingleNode()` and so on. I can do that but it will just about refactoring all the calls. What do you think?

Reply:
ah then no, I didn't see that part